src/core/pointcloud/qgslazinfo.h

```diff
@@ -0,0 +1,114 @@
+/***************************************************************************
+                         qgslazinfo.h
+                         --------------------
+    begin                : April 2022
+    copyright            : (C) 2022 by Belgacem Nedjima
+    email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSLAZINFO_H
+#define QGSLAZINFO_H
+
+#include <fstream>
+
+#include "qgspointcloudblock.h"
+#include "qgspointcloudattribute.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgspointcloudindex.h"
+
+#include "lazperf/lazperf.hpp"
+#include "lazperf/readers.hpp"
+
+#define SIP_NO_FILE
+
+class CORE_EXPORT QgsLazInfo
+{
```

Comment:
One thing that is confusing about this class is that most of the code is there only for COPC support - there are various assumptions that the LAZ file is 1.4 (point record 6 / 7 / 8), but then there is parseExtrabytes() function that is used also for EPT, which can contain LAZ 1.2.

My suggestion is to rename this class e.g. to `QgsCopcHeaderInfo` and keep it limited to COPC functionality. Anything that is meant for both EPT and COPC should probably go back to QgsLazDecoder - mainly parseExtrabytes(), not sure if there's anything else.

Reply:
COPC is just LAZ and all things that we are doing in this class are common between the two. So instead of splitting the handling of which attributes exist in the dataset between the two, it should be unified IMO. The only important thing that COPC has over standard LAZ is a VLR that points to the hierarchy root and that is a relatively small portion of code that can be handled in the point cloud index.
The way we handle EPT header info feels a bit complicated because we never actually read the original file and we don't parse its header data (I mean the original laz not ept.json). What we are doing with untwine in a sense is just splitting the original dataset file into many laz files and in my understanding the header data that is taken from the ept.json file shouldn't be used to read the tiles in the first place. EPT is in kinda agnostic to the tiles format (the individual tiles could be binary, zstd or laz). I am suggesting that we keep it as QgsLazInfo and this class should be used for decoding LAZ data because we actually won't need any parameters from EPT to read a LAZ tile correctly.

Sorry for the long comment 😅 . We can discuss it in more details tommorow and I will probably have a better code for this :)